### ISSUE TITLE
Julia Loader

### ISSUE THREAD

**Issue by viferga** [metacall/core#135]
## 🚀 Feature

This is an issue for writing all necessary information for the Julia loader. I will be editing this during the day explaining the details, we can use this to talk about any doubts, design decisions or implementation details.

###  Proof of Concept
The proof of concept done by Nikhil Narang (I don't know his GitHub account yet) can be found here:
[5_6114092993344439109.zip](https://github.com/metacall/core/files/6474523/5_6114092993344439109.zip)

For compiling it:
```sh
gcc -o test -fPIC -I/Applications/Julia-1.6.app/Contents/Resources/julia/include/julia -L/Applications/Julia-1.6.app/Contents/Resources/julia/lib -Wl,-rpath,/Applications/Julia-1.6.app/Contents/Resources/julia/lib julia.c -ljulia
```

**Comment by viferga** [comment]
Here's the base implementation for the loader: https://github.com/metacall/core/commit/cd015706c0e4c2d05e648460b7a1b99b796b61b0

**Comment by prithvi009** [comment]
Can I work on this issue?

**Comment by giarve** [comment]
> Can I work on this issue?

Absolutely, go ahead.

**Comment by ashutosh887** [comment]
@giarve Should I work on this Issue if it's still not resolved?
Please assign it to me

**Comment by giarve** [comment]
> @giarve Should I work on this Issue if it's still not resolved? Please assign it to me

Yes you can. Open a pull request as "draft" once you have some code to show that you are working on it.

**Comment by ashutosh887** [comment]
Thanks a lot @giarve 
I'll start working on it

**Comment by Rohxn16** [comment]
@giarve 
Hey man, I think it's been a while since the last request...do you mind if I work on this if it's still open?

**Comment by c1pher1005** [comment]
Hey, i am interested in working on this issue could you please assign me on this?

**Comment by viferga** [comment]
For anybody asking, you can still work on this @c1pher1005 @Rohxn16 @ashutosh887 

If you have doubts about this feel free to ask.
